### PR TITLE
TWE-2 - BE - Add division signpost block

### PR DESCRIPTION
**Pull request by SharmaineLim**
[Link to Ticket](https://torchbox.atlassian.net/browse/TWE-2)

### Description of Changes Made

This ticket adds a new block for signposting to division pages from the homepage.

FE styling covered by #323 

### How to Test

1. Edit the homepage.
2. Try to add a new block in the Body field. You should see "Division signpost" as a block.
3. Adding the "Division signpost" block and filling in the required fields should display the new content on the preview.

### Screenshots

<details><summary>Editor</summary>

image

</details>

### MR Checklist

- [ ] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [ ] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Updated field spec (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [x] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Light and dark mode

- [ ] I have tested the changes in both light and dark mode
- [x] The change is not relevant to dark and light mode

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/core/blocks.py [hunk 1]
```diff
@@ -297,6 +298,50 @@ class IconChoice(models.TextChoices):
     WAGTAIL = "wagtail", "wagtail icon"
 
 
+class DivisionSignpostCardValue(blocks.StructValue):
+    def get_heading(self):
+        if override := self.get("heading"):
+            return override
+        return self["page"].title
```
Comment (ababic) [inline tbx/core/blocks.py:305]: Would this be simpler to do in `DivisionSignpostCardBlock.get_context()`?

Comment (ababic) [reply tbx/core/blocks.py:305]: Or even a `{% firstof card.heading card.page.title %}` in the template

Comment (SharmaineLim) [reply tbx/core/blocks.py]: Thanks, I've tweaked it to use `firstof` instead. I've gotten too used to using `StructValue`!

File: tbx/core/blocks.py [hunk 2]
```diff
@@ -297,6 +298,42 @@ class IconChoice(models.TextChoices):
     WAGTAIL = "wagtail", "wagtail icon"
 
 
+class DivisionSignpostCardBlock(blocks.StructBlock):
+    class ColourTheme(models.TextChoices):
+        CORAL = "theme-coral", "Coral"
```
Comment (helenb) [inline tbx/core/blocks.py:303]: Just to check: is this the new theme naming for the whole site? Replacing banana etc?

Comment (SharmaineLim) [reply tbx/core/blocks.py:303]: This block has a reduced number of available colours compared to the rest of the site.

The theme colours that will be available to the site are: coral, nebuline, lagoon and green, but for this block, only coral, nebuline and lagoon have been designed (for charity, public sector and wagtail respectively), so I made it its own subclass of colour options.

**Comment by SharmaineLim** [comment]
> I noticed that the heading for each card isn't a required field, but it should be.

I didn't make it required because if it's not populated, we can just use the one from the page. (I was planning to do the same for description & image, but then I realised that this project doesn't have listing summaries and listing images.)

**Pull request merged**